sql/core/src/main/java/org/apache/spark/sql/execution/UnsafeFixedWidthAggregationMap.java

```diff
@@ -226,10 +226,10 @@ public void free() {
   }
 
   /**
-   * Gets the average hash map probe per looking up for the underlying `BytesToBytesMap`.
+   * Gets the collision rate per looking up for the underlying `BytesToBytesMap`.
    */
-  public double getAverageProbesPerLookup() {
-    return map.getAverageProbesPerLookup();
+  public double getHashCollisionsPerLookup() {
+    return map.getAverageProbesPerLookup() - 1.0;
```

Comment:
Why `- 1.0` here? This change is different from the description?

Reply:
IIRC this metric was added after a bug was found and fixed that caused a lot of hash conflicts.
If a collision list is of length there are no hash conflicts, so the number of "collisions" seems better defined as length of list minus 1 (so that a list of length 2 is one conflict)...

Hm. What about getting rid of the -1 and naming it "avg hash bucket list length"?

Reply:
Edit: "avg hash probe bucket list iterations", because that's exactly what it is...

Reply:
To make Spark users understand SQL metrics more, IMO it'd be better to describe more about them in documents instread of variable names. I'm not sure about who is the target of this pr, deverlopres or users though...

Reply:
advanced users. :)

Reply:
ah, ok!